source/core/index.ts

```diff
@@ -647,9 +647,14 @@ export default class Request extends Duplex implements RequestEvents<Request> {
 				options.searchParams = new URLSearchParams(options.searchParams as Record<string, string>);
 
 				// `normalizeArguments()` is also used to merge options
-				defaults?.searchParams?.forEach((value, key) => {
-					(options!.searchParams as URLSearchParams).append(key, value);
-				});
+				if (defaults?.searchParams) {
+					new URLSearchParams(defaults.searchParams).forEach((value, key) => {
+						const searchParameters = options!.searchParams as URLSearchParams;
+						if (!searchParameters.get(key)) {
+							searchParameters.append(key, value);
+						}
+					});
+				}
```

Comment:
This is exactly the same as:

```ts
				defaults?.searchParams?.forEach((value, key) => {
					(options!.searchParams as URLSearchParams).set(key, value);
				});
```

Reply:
Not exactly. If you pass a `string` via the init hook then it will dump for example.

Reply:
You would need to change the docs then

Reply:
> Not exactly. If you pass a string via the init hook then it will dump for example.

A test would be awesome :)

Reply:
Sure... though, is there any magic about the tests? I just noticed that I can even use `t.is(true, false);` and the test does not fail locally when using `yarn test test/hooks.ts`.

Reply:
And is there any way to output debug information during tests?